### PR TITLE
Removing bonus points from teacher application

### PR DESCRIPTION
**Pull request by hacodeorg**
# Description
Bonus points are no longer part of the 20-21 rubric. We had removed them from application detail view but still calculate them on the server side. This change will remove bonus points from server-side teacher 20-21 application code. 

This change doesn't remove `bonus_points` and `total_score` in [application_serializer.rb](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/serializers/api/v1/pd/application_serializer.rb) and [detail_view_content.jsx](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/pd/application_dashboard/detail_view_contents.jsx) because the code are shared among multiple application years and application types (teacher or facilitator).

## Links
- [jira](https://codedotorg.atlassian.net/browse/PLC-507)

## Testing story
- Manual test
  - Creates a teacher application at `http://localhost-studio.code.org:3000/pd/application/teacher`. Then verifies that application is displayed correctly in detail view `https://studio.code.org/pd/application_dashboard/csd_teachers/<application_id>`

- Automated tests
  - Unit tests
    - bundle exec rails test test/models/pd/application/teacher2021_application_test.rb
    - bundle exec rails test test/controllers/api/v1/pd/application/teacher_applications_controller_test.rb
    - bundle exec rails test test/controllers/api/v1/pd/applications_controller_test.rb
    - bundle exec rails test test/serializers/api/v1/pd/application_serializer_test.rb
  - Eyes test
    - ./runner.rb -d localhost-studio.code.org:3000 --html --eyes -c Chrome -f features/pd/teacher_application_dashboard.feature
    - ./runner.rb -d localhost-studio.code.org:3000 --html --eyes -c Chrome -f features/pd/teacher_application.feature

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by agealy — APPROVED** [review]

File: dashboard/app/models/pd/application/teacher2021_application.rb [hunk 1]
```diff
@@ -877,7 +871,7 @@ def meets_scholarship_criteria
 
     # @override
     def total_score
```
Comment (bencodeorg) [inline dashboard/app/models/pd/application/teacher2021_application.rb:873]: More for my understanding than anything...`total_score` is still necessary here?

Comment (bencodeorg) [reply dashboard/app/models/pd/application/teacher2021_application.rb]: Never mind, see comment your description now!

Comment (bencodeorg) [reply dashboard/app/models/pd/application/teacher2021_application.rb:873]: Never mind, see your comment in the description!

**Review by islemaster — APPROVED** [review]
:+1: Nice cleanup.

**Pull request merged**